enhancements/installer/assisted-installer.md

```diff
@@ -0,0 +1,320 @@
+---
+title: assisted-installer
+authors:
+  - "@avishayt"
+  - "@hardys"
+  - "@dhellmann"
+reviewers:
+  - "@beekhof"
+  - "@deads2k"
+  - "@hexfusion"
+  - "@mhrivnak"
+approvers:
+  - "@crawford"
+  - "@abhinavdahiya"
+  - "@eparis"
+creation-date: 2020-06-09
+last-updated: 2020-06-10
+status: implementable
+see-also:
+  - "/enhancements/baremetal/minimise-baremetal-footprint.md"
+---
+
+# Assisted Installer for Connected Environments
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement describes changes in and around the installer to
+assist with deployment on user-provisioned infrastructure. The use
+cases are primarily relevant for bare metal, but in the future may be
+applicable to cloud users who are running an installer UI directly
+instead of using a front-end such as `cloud.redhat.com` or the UI
+provided by their cloud vendor.
+
+## Motivation
+
+The target user is someone wanting to deploy OpenShift, especially on
+bare metal, with as little up-front infrastructure dependencies as
+possible. This person has access to servers and wants to run workloads
+quickly. They do not necessarily have the administrative privileges to
+create private VLANs, configure DHCP/PXE servers, or manage other
+aspects of the infrastructure surrounding the hardware where the
+cluster will run. If they do have the required privileges, they may
+not want to delegate them to the OpenShift installer for an IPI
+installation, preferring instead to use their existing tools and
+processes for some or all of that configuration. They are willing to
+accept that the cluster they build may not have all of the same
+features as an IPI cluster on day 1, but that by taking additional
+steps later they will be able to add more automatic infrastructure
+management and make it fully production-ready.
+
+### Goals
+
+- Make initial deployment of usable and supportable clusters simpler.
+- Move more infrastructure configuration from day 1 to day 2.
+- Support connected on-premise deployments.
+- Support existing IPI features, especially for day 2 cluster
+  management and scale-out.
+
+### Non-Goals
+
+- Because the initial focus is on bare metal, this enhancement does
+  not exhaustively cover variations needed to offer similar features
+  on other platforms (such as changes to image formats, the way a host
+  boots, etc.). It is desirable to support those platforms, but that
+  work will be described separately.
+- Support environments where hosts can reach the internet unimpeded
```

Comment:
I think some element of Alex's comment would be good to include here

https://github.com/dhellmann/openshift-enhancements/pull/1#discussion_r439120554

i.e. to set clear expectations that "the disconnected solution isn't just going to be us scrambling to try to package this hosted service up into something that can run on-prem".

May as well include a little commentary along these for the non-goal, like you do for the first non-goal.

Reply:
Good point. I've updated this bullet point.